AEPCore/Sources/core/MobileCore.swift

```diff
@@ -38,6 +39,40 @@ public final class MobileCore: NSObject {
     /// Pending extensions to be registered for legacy support
     static var pendingExtensions = ThreadSafeArray<Extension.Type>(identifier: "com.adobe.pendingExtensions.queue")
 
+    public static func start(with appID: String,
+                             logLevel: LogLevel? = nil,
+                             configUpdates: [String: Any]? = nil,
+                             additionalContextData: [String: Any]? = nil,
+                             disableLifecycleStart: Bool? = nil,
+                             applicationState: UIApplication.State? = nil
+                             ) {
+        if let logLevel = logLevel {
+            setLogLevel(logLevel)
+        }
+        // Register Extensions, call configureWithAppId from callback
+        let classList = ClassFinder.classes(conformToProtocol: Extension.self)
```

Comment:
I feel we move this inside Dispatch queue so that it does not block the main thread?